File: scripts/Fib.dart
import 'package:dart_godot/dart_godot.dart';

 class Fib extends Node {
   
   double fib (double n){
     if (n <= 1){
       return n;
     }else{
       return fib(n-1)+fib(n-2);
     }
   }

  @override
  void ready() {
    super.ready();
    print(fib(10));
    print(fib(20));
    print(fib(40));
    print(fib(80));
  }

  @override
  void process(double delta) {
    super.process(delta);
    //print('the dart is on :) ${delta}');
  }
}

void main() {
  var fib = Fib();
}

File: compiled/Fib.cs
using Godot;

public class Fib : Godot.Node
{
	private DartConnector _dartConnector;
	
	public override void _EnterTree()
	{
		base._EnterTree();
		_dartConnector = new DartConnector("DartGodot\\compiled\\data\\Fib.d2g");
		_dartConnector.EnterTree();
	}
	public override void _Ready()
	{
		_dartConnector.Ready();
	}
	public override void _Process(float delta) 
	{ 
		_dartConnector.Process(delta);
	}
	public override void _PhysicsProcess(float delta) 
	{ 
		base._PhysicsProcess(delta); 
		_dartConnector.PhysicsProcess(delta);
	}
	public override void _ExitTree() 
	{ 
		base._ExitTree(); 
		_dartConnector.ExitTree();
	}
}
